app/lib/cashu/schemas.ts

```diff
@@ -0,0 +1,32 @@
+import { z } from 'zod';
+import {
+  type NUT10Secret,
+  type NUT10SecretTag,
+  WELL_KNOWN_SECRET_KINDS,
+  type WellKnownSecretKind,
+} from './types';
+
+export const NUT10SecretTagSchema = z
+  .array(z.string())
+  .nonempty()
+  .refine(
+    (arr): arr is [string, ...string[]] => arr.length >= 1,
+  ) satisfies z.ZodType<NUT10SecretTag>;
```

Comment:
we still have the type defined manually. with zod you can do this in types.ts:
```
type NUT10SecretTag = z.infer<typeof NUT10SecretTagSchema >;
```
and then lose this `satisfies` thing. wouldn't that work?

Reply:
open secret guys are also defining the schema and inferring the type from the schema. see here https://github.com/OpenSecretCloud/OpenSecret-SDK/blob/master/src/lib/attestation.ts#L13

Reply:
Yea that works, should I move the schemas to types?

Reply:
yeah we can keep it in the same file

Reply:
moved schemas to types.ts and inferred NUT10SecretTag and RawNUT10Secret with zod